### PR TITLE
Fix new possible nullref inspection due to delegate initialisation in constructor

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

File: osu.Game/Screens/Play/GameplayMenuOverlay.cs [hunk 1]
```diff
@@ -142,6 +140,8 @@ private void load(OsuColour colours)
                 },
             };
 
+            State.ValueChanged += s => InternalButtons.Deselect();
```
Comment (smoogipoo) [inline osu.Game/Screens/Play/GameplayMenuOverlay.cs:143]: Any reason for moving into here instead of `LoadComplete()`?

Comment (peppy) [reply osu.Game/Screens/Play/GameplayMenuOverlay.cs:143]: Just for safety... `VisibilityContainer` still has very weird state handling logic that I was preferring not to drill into (I know it does some kind of state change handling/allowance for `InitialState` changing in BDL or similar...)

Comment (smoogipoo) [reply osu.Game/Screens/Play/GameplayMenuOverlay.cs:143]: Mmmk. I'm 95% expecting this to fail in some edge case with some component trying to make the overlay visible and it dying on a transform-from-non-load-thread exception. But for now I'll :see_no_evil:

**Review by smoogipoo — APPROVED** [review]

File: osu.Game/Screens/Play/GameplayMenuOverlay.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Screens/Play/GameplayMenuOverlay.cs:143]: Well it's worked until now (such a case would have already been hitting a nullref). We can move it if you want, but please test at your end if so 😅

**Pull request merged**